src/DynamoUtilities/PIIDetector.cs

```diff
@@ -0,0 +1,102 @@
+    public static class PIIDetector
+    {
+        /// <summary>
+        /// Removes the PII data from a JSON workspace
+        /// </summary>
+        /// <param name="jsonObject"></param>
+        /// <returns></returns>
+        public static Tuple<string, JObject> RemovePIIData(JObject jsonObject)
+        {
+            JObject jObjectResult = jsonObject;
+            string uuid = string.Empty;
+
+            foreach (var properties in jObjectResult.Properties())
+            {
+                if (properties.Name == "Uuid")
+                {
+                    uuid = properties.Value.ToString();
+                }
+                else if (properties.Name == "Nodes")
+                {
+                    if (properties.Value.Type == JTokenType.Array)
+                    {
+                        var nodes = (JArray)properties.Value;
+                        foreach (JObject node in nodes)
+                        {
+                            var nodeProperties = node.Children<JProperty>();
+
+                            JProperty inputValue = nodeProperties.FirstOrDefault(x => x.Name == "InputValue");
+                            JProperty hintPath = nodeProperties.FirstOrDefault(x => x.Name == "HintPath");
+                            JProperty code = nodeProperties.FirstOrDefault(x => x.Name == "Code");
+
+                            if (inputValue != null) { inputValue.Value = RemovePIIData((string)inputValue.Value); }
+                            if (hintPath != null) { hintPath.Value = RemovePIIData((string)hintPath.Value); }
+                            if (code != null) { code.Value = RemovePIIData((string)code.Value); }
+                        }
+                    }
+                }
+                else if (properties.Name == "View")
+                {
+                    var view = (JObject)properties.Value;
+                    var viewProperties = view.Children<JProperty>();
+                    var annotations = viewProperties.FirstOrDefault(x => x.Name == "Annotations");
+
+                    if (annotations != null)
+                    {
+                        if (annotations.Value.Type == JTokenType.Array)
+                        {
+                            var annotationsList = (JArray)annotations.Value;
+
+                            foreach (JObject annotation in annotationsList)
+                            {
+                                var annotationProperties = annotation.Children<JProperty>();
+                                JProperty title = annotationProperties.FirstOrDefault(x => x.Name == "Title");
+                                if (title != null) { title.Value = RemovePIIData((string)title.Value); }
+                            }
+                        }
+                    }
+                }
+            }
+            return new Tuple<string, JObject>(uuid, jObjectResult);
+        }
+
+        static string emailPattern = @"\w+([-+.]\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*";
```

Comment:
is this best practice? where are these regexes pulled from? I don't know how others feel, but this seems very hard to understand, reason about or maintain - especially because there are no tests in this PR.

Reply:
of note
https://docs.aws.amazon.com/comprehend/latest/dg/how-pii.html
https://microsoft.github.io/presidio/analyzer/

Reply:
> of note https://docs.aws.amazon.com/comprehend/latest/dg/how-pii.html https://microsoft.github.io/presidio/analyzer/

I have added an Unit Test for it so We can discuss in detail the used regexes and their scope / granularity, I have already used them in the past for others projects (As we know this feature is cross project/company).

About your references I have already considered similar options in the Spike from this task : https://jira.autodesk.com/browse/DYN-5964 . I considered the PII Helper class as the main option but its opened to be extended or changed as the Team decide.